### ISSUE TITLE
Slider bug: handler can only be used once

### ISSUE THREAD

**Issue by aboveyunhai** [chakra-ui/chakra-ui#3040]
# 🐛 Bug report

slider can only be moved once 

## 💥 Steps to reproduce

 Slider is broken/fixed after first slide

## 💻 Link to reproduction

 It seems like all sliders on official Chakra doc has the exact problem.
 https://chakra-ui.com/docs/form/slider

## 🧐 Expected behavior
Slider can be freely moved without customization.

## 📝 Additional information
I saw this error once in dev before I found the same issue persist on Chakra-UI doc but cannot reproduce.
This may not relate to the problem.
`Ignored attempt to cancel a touchstart event with cancelable=false, for example because scrolling is in progress and cannot be interrupted.`

**Comment by ItsWendell** [comment]
I was experiencing the exact same issue and already wondered why nobody opened an issue for this, if anyone has a temporary workaround, would love to know!

**Comment by Diveafall** [comment]
Having the same issue here. Moving the slider using the keyboard works, but it re-renders the parent component for no reason.

**Comment by j-dominguezp** [comment]
I had to downgrade to v1.1.2 to have this working back.

**Comment by delucca** [comment]
any updates on this? I would like to have this issue fixed since `1.1.2` has bug #2956 and `1.1.4` has this bug. Both of those are really troubling for my app

**Comment by with-heart** [comment]
I've tracked this issue down to this change: https://github.com/chakra-ui/chakra-ui/commit/27c8a6ce627aa18e52998b68ede6661e4c117be7#diff-28915326605476712d3e02282e5a571ea49369d7bd59eb52f2ee104cf9077bfcL24-L31

Where we were previously creating a separate `listener` function that is added and removed inside the `useEffect`, we're now directly adding and removing the `fn` itself which leaves us in a state where we no longer have a listener after the first event happens.

**Comment by tomhicks** [comment]
> I was experiencing the exact same issue and already wondered why nobody opened an issue for this, if anyone has a temporary workaround, would love to know!

I have a workaround...

```
const [key, setKey] = useState(0)
React.useEffect(() => {
  function handleMouseUp() {
    setKey(Math.random())
  }

  document.addEventListener("mouseup", handleMouseUp)
  
  return () => {
    document.removeEventListener("mouseup", handleMouseUp)
  }
})
<Slider
  key={key}
/>
```

**Comment by delucca** [comment]
Taking a look at #3056 it seems the issue was fixed, they just need to publish a new release.

**Comment by tomhicks** [comment]
Yeah I saw but it's not released yet so a workaround is still useful

**Comment by dodas** [comment]
Released now in `@chakra-ui/react@v1.1.5`.🎉
